src/test/project-manager.local.test.ts

```diff
@@ -0,0 +1,44 @@
+import * as fs from 'mz/fs';
+import * as path from 'path';
+import * as rimraf from 'rimraf';
+import * as temp from 'temp';
+import { FileSystemUpdater, LocalFileSystem } from '../fs';
+import { InMemoryFileSystem } from '../memfs';
+import { ProjectManager } from '../project-manager';
+import { setStrict, toUnixPath } from '../util';
+import { describeProjectManager, expectedFiles } from './project-manager-helpers';
+
+describe('Local ProjectManager', () => {
```

Comment:
A unit test should only test the unit under test and should assume all dependencies (like the passed FileSystem interface) behave correctly. We shouldn't test every possible combination, or we get into an n x n problem. There should be one unit test suite for ProjectManager that tests it behaves correctly given a correctly-behaving file system (like the MapFileSystem) and one suite for e.g. LocalFileSystem that tests it behaves correctly too.